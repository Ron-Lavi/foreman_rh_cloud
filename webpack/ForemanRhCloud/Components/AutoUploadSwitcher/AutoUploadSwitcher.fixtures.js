export const handleToggleResponse = { data: { autoUploadEnabled: true } };
