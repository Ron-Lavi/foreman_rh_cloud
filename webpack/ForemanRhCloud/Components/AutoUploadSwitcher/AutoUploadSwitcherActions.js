import API from 'foremanReact/API';
import {
  AUTO_UPLOAD_TOGGLE,
  AUTO_UPLOAD_TOGGLE_ERROR,
} from './AutoUploadSwitcherConstants';

export const handleToggle = currentAutoUploadEnabled => async dispatch => {
  const toggledAutoUploadEnabled = !currentAutoUploadEnabled;
  try {
    const {
      data: { autoUploadEnabled },
    } = await API.post('auto_upload', { value: toggledAutoUploadEnabled });
    dispatch({
      type: AUTO_UPLOAD_TOGGLE,
      payload: {
        autoUploadEnabled,
      },
    });
  } catch (error) {
    dispatch({
      type: AUTO_UPLOAD_TOGGLE_ERROR,
      payload: {
        error: error.message,
      },
    });
  }
};
