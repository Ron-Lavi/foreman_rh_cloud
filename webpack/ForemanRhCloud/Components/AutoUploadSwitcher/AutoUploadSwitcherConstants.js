export const AUTO_UPLOAD_TOGGLE = 'AUTO_UPLOAD_TOGGLE';
export const AUTO_UPLOAD_TOGGLE_ERROR = 'AUTO_UPLOAD_TOGGLE_ERROR';
