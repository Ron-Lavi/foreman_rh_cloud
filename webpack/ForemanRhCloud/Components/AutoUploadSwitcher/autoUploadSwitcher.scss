.auto_upload_switcher {
  float: right;
}
