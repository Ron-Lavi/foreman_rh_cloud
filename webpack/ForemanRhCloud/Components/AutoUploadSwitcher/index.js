import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './AutoUploadSwitcherActions';
import AutoUploadSwitcher from './AutoUploadSwitcher';
import { selectAutoUploadEnabled } from '../AccountList/AccountListSelectors';

// map state to props
const mapStateToProps = state => ({
  autoUploadEnabled: selectAutoUploadEnabled(state),
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export connected component
export default connect(mapStateToProps, mapDispatchToProps)(AutoUploadSwitcher);
