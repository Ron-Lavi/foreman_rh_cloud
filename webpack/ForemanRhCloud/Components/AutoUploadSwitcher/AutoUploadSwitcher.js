import React from 'react';
import PropTypes from 'prop-types';
import { Switch, FieldLevelHelp } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';
import './autoUploadSwitcher.scss';

const AutoUploadSwitcher = ({ autoUploadEnabled, handleToggle }) => (
  <div className="auto_upload_switcher">
    <Switch
      size="mini"
      value={autoUploadEnabled}
      onChange={() => handleToggle(autoUploadEnabled)}
    />
    <FieldLevelHelp
      content={__(
        'Enable automatic upload of your host inventory to the Red Hat cloud'
      )}
    />
    <span>Allow Auto Upload</span>
  </div>
);

AutoUploadSwitcher.propTypes = {
  autoUploadEnabled: PropTypes.bool,
  handleToggle: PropTypes.func.isRequired,
};

AutoUploadSwitcher.defaultProps = {
  autoUploadEnabled: true,
};

export default AutoUploadSwitcher;
