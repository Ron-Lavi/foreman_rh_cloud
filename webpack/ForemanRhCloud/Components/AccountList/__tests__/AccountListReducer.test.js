import { testReducerSnapshotWithFixtures } from '@theforeman/test';

import {
  INVENTORY_ACCOUNT_STATUS_POLLING,
  INVENTORY_ACCOUNT_STATUS_POLLING_START,
  INVENTORY_ACCOUNT_STATUS_POLLING_STOP,
  INVENTORY_ACCOUNT_STATUS_POLLING_ERROR,
  INVENTORY_PROCESS_RESTART,
} from '../AccountListConstants';
import reducer from '../AccountListReducer';
import {
  accounts,
  error,
  pollingProcessID,
  accountID,
  processStatusName,
  autoUploadEnabled,
} from '../AccountList.fixtures';
import { AUTO_UPLOAD_TOGGLE } from '../../AutoUploadSwitcher/AutoUploadSwitcherConstants';

const fixtures = {
  'should return the initial state': {},
  'should handle INVENTORY_ACCOUNT_STATUS_POLLING': {
    action: {
      type: INVENTORY_ACCOUNT_STATUS_POLLING,
      payload: {
        accounts,
        autoUploadEnabled,
      },
    },
  },
  'should handle INVENTORY_ACCOUNT_STATUS_POLLING_ERROR': {
    action: {
      type: INVENTORY_ACCOUNT_STATUS_POLLING_ERROR,
      payload: { error },
    },
  },
  'should handle INVENTORY_ACCOUNT_STATUS_POLLING_START': {
    action: {
      type: INVENTORY_ACCOUNT_STATUS_POLLING_START,
      payload: {
        pollingProcessID,
      },
    },
  },
  'should handle INVENTORY_ACCOUNT_STATUS_POLLING_STOP': {
    action: {
      type: INVENTORY_ACCOUNT_STATUS_POLLING_STOP,
    },
  },
  'should handle INVENTORY_PROCESS_RESTART': {
    action: {
      type: INVENTORY_PROCESS_RESTART,
      payload: {
        accountID,
        processStatusName,
      },
    },
  },
  'should handle AUTO_UPLOAD_TOGGLE': {
    action: {
      type: AUTO_UPLOAD_TOGGLE,
      payload: {
        autoUploadEnabled,
      },
    },
  },
};

describe('AccountList reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
