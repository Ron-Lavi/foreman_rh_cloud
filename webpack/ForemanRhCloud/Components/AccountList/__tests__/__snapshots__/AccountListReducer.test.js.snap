// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountList reducer should handle AUTO_UPLOAD_TOGGLE 1`] = `
Object {
  "accounts": Object {},
  "autoUploadEnabled": true,
  "error": null,
  "pollingProcessID": 0,
}
`;

exports[`AccountList reducer should handle INVENTORY_ACCOUNT_STATUS_POLLING 1`] = `
Object {
  "accounts": Object {
    "Account1": Object {
      "generate_report_status": "running",
      "label": "test_org1",
      "upload_report_status": "running",
    },
    "Account2": Object {
      "generate_report_status": "failure",
      "label": "test_org2",
      "upload_report_status": "unknown",
    },
    "Account3": Object {
      "generate_report_status": "running",
      "label": "test_org3",
      "upload_report_status": "success",
    },
  },
  "autoUploadEnabled": true,
  "error": null,
  "pollingProcessID": 0,
}
`;

exports[`AccountList reducer should handle INVENTORY_ACCOUNT_STATUS_POLLING_ERROR 1`] = `
Object {
  "accounts": Object {},
  "error": "some-error",
  "pollingProcessID": 0,
}
`;

exports[`AccountList reducer should handle INVENTORY_ACCOUNT_STATUS_POLLING_START 1`] = `
Object {
  "accounts": Object {},
  "error": null,
  "pollingProcessID": 0,
}
`;

exports[`AccountList reducer should handle INVENTORY_ACCOUNT_STATUS_POLLING_STOP 1`] = `
Object {
  "accounts": Object {},
  "error": null,
  "pollingProcessID": 0,
}
`;

exports[`AccountList reducer should handle INVENTORY_PROCESS_RESTART 1`] = `
Object {
  "accounts": Object {
    "user@redhat.com": Object {
      "upload_report_status": "Restarting...",
    },
  },
  "error": null,
  "pollingProcessID": 0,
}
`;

exports[`AccountList reducer should return the initial state 1`] = `
Object {
  "accounts": Object {},
  "error": null,
  "pollingProcessID": 0,
}
`;
